platform-hub-api/app/controllers/kubernetes/tokens_controller.rb

```diff
@@ -65,6 +65,43 @@ def destroy
     end
   end
 
+  # POST /kubernetes/tokens/:user_id/:cluster/escalate
+  def escalate
+    authorize! :manage, :identity
+
+    privileged_group, expires_in_secs = params.require([:privileged_group, :expires_in_secs])
+
+    tokens, privileged_token = Kubernetes::TokenService.escalate_token(
+      identity_data,
+      params[:cluster],
+      privileged_group,
+      expires_in_secs
+    )
+
+    unless privileged_token.valid?
+      render_model_errors privileged_token.errors and return
+    end
+
+    @identity.with_lock do
+      @identity.data[:tokens] = tokens
+
+      if @identity.save!
+        AuditService.log(
+          context: audit_context,
+          action: 'escalate_kubernetes_token',
+          auditable: @identity,
+          data: { cluster: params[:cluster] },
```

Comment:
Is it worth putting the group + timing requested here?

Reply:
not crucial as it's part of description but sure we could add that in.

Reply:
Yeah, just thinking in terms of writing a query to get all audits for escalation using a particular group.